File: Stack/StaticStack/StaticStack.py
class StaticStack:
    def __init__(self, capacity=5):
        self.capacity = capacity
        self.stack = [None] * capacity
        self.top_idx = -1  # Indică indexul elementului de sus

    def isFull(self):
        return self.top_idx == self.capacity - 1

    def isEmpty(self):
        return self.top_idx == -1

    def push(self, value):
        if self.isFull():
            print("Stack Overflow! Cannot push", value)
            return False
        self.top_idx += 1
        self.stack[self.top_idx] = value
        return True

    def pop(self):
        if self.isEmpty():
            print("Stack Underflow! Cannot pop")
            return None
        value = self.stack[self.top_idx]
        self.stack[self.top_idx] = None  
        self.top_idx -= 1
        return value

    def peek(self):
        if self.isEmpty():
            print("Stack is empty!")
            return None
        return self.stack[self.top_idx]

    def size(self):
        return self.top_idx + 1

    def print(self):
        if self.isEmpty():
            print("Stack is empty")
            return
        print("Stack (Top -> Bottom):")
        for i in range(self.top_idx, -1, -1):
            print(f"| {self.stack[i]} |")
        print("-------")

if __name__ == "__main__":
    stack = StaticStack(3)
    stack.push(10)
    stack.push(20)
    stack.push(30)
    stack.push(40)  # Overflow
    stack.print()

    print("Popped:", stack.pop())
    print("Top element:", stack.peek())
    stack.print()

File: Stack/StaticStack/StaticStack.java
public class StaticStack {
    private int[] stack;
    private int top;
    private int capacity;

    public StaticStack(int capacity) {
        this.capacity = capacity;
        this.stack = new int[capacity];
        this.top = -1;
    }

    public boolean isFull() {
        return top == capacity - 1;
    }

    public boolean isEmpty() {
        return top == -1;
    }

    public boolean push(int value) {
        if (isFull()) {
            System.out.println("Stack Overflow! Cannot push " + value);
            return false;
        }
        stack[++top] = value;
        return true;
    }

    public int pop() {
        if (isEmpty()) {
            System.out.println("Stack Underflow! Cannot pop");
            return -1;
        }
        return stack[top--];
    }

    public int peek() {
        if (isEmpty()) {
            System.out.println("Stack is empty!");
            return -1;
        }
        return stack[top];
    }

    public int size() {
        return top + 1;
    }

    public void print() {
        if (isEmpty()) {
            System.out.println("Stack is empty");
            return;
        }
        System.out.println("Stack (Top -> Bottom):");
        for (int i = top; i >= 0; i--) {
            System.out.println("| " + stack[i] + " |");
        }
        System.out.println("-------");
    }

    public static void main(String[] args) {
        StaticStack stack = new StaticStack(5);
        stack.push(10);
        stack.push(20);
        stack.push(30);
        stack.print();

        System.out.println("Popped: " + stack.pop());
        System.out.println("Top element: " + stack.peek());
        stack.print();
    }
}
